tools/clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -1817,9 +1818,36 @@ Value *ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
     return Builder.CreateExtractElement(Visit(E), (uint64_t)0);
   }
   case CK_FlatConversion: {
-      llvm::Value *val = Visit(E);
-      llvm::Value *elem = Builder.CreateExtractValue(val, (uint64_t)0);
-      return EmitScalarConversion(elem, E->getType(), DestTy);
+    llvm::Value *Src = Visit(E);
+
+    // We should have a compound type (struct or array) on one side,
+    // and a numeric type (scalar, vector or matrix) on the other.
```

Comment:
Does this mean aggregate to aggregate (or compound) cast is handled elsewhere?

Reply:
Yes, in `CGExprAgg`